static/app/components/slot/slot.spec.tsx

```diff
@@ -0,0 +1,123 @@
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {Text} from 'sentry/components/core/text';
+
+import {Slot} from './slot';
+
+describe('Slot', () => {
+  describe('production environment', () => {
+    let originalEnv: string | undefined;
+
+    beforeEach(() => {
+      originalEnv = process.env.NODE_ENV;
+      process.env.NODE_ENV = 'production';
+    });
+
+    afterEach(() => {
+      process.env.NODE_ENV = originalEnv;
+    });
+
+    it('renders null for invalid children', () => {
+      render(<Slot>Hello</Slot>);
+      expect(screen.queryByText('Hello')).not.toBeInTheDocument();
+      render(<Slot>{['Hello', 'from', 'production']}</Slot>);
+      expect(screen.queryByText('Hello')).not.toBeInTheDocument();
+    });
+
+    it('renders null if multiple children are provided', () => {
+      render(
+        <Slot>
+          <Text>Hello</Text>
+          <Text>World</Text>
+        </Slot>
+      );
+      expect(screen.queryByText('Hello')).not.toBeInTheDocument();
+      expect(screen.queryByText('World')).not.toBeInTheDocument();
+    });
+
+    it('renders the child if it is a valid element', () => {
+      render(
+        <Slot>
+          <Text>Hello</Text>
+        </Slot>
+      );
+      expect(screen.getByText('Hello')).toBeInTheDocument();
+    });
+  });
+
+  describe('development environment', () => {
+    let originalEnv: string | undefined;
+
+    beforeEach(() => {
+      originalEnv = process.env.NODE_ENV;
+      process.env.NODE_ENV = 'development';
+    });
+
+    afterEach(() => {
+      process.env.NODE_ENV = originalEnv;
+    });
+
+    it('throws an error if children is not a valid element', () => {
+      expect(() => render(<Slot>{['Hello', 'from', 'development']}</Slot>)).toThrow(
+        'React.Children.only expected to receive a single React element child.'
+      );
+    });
+
+    it('throws an error if children count is greater than 1', () => {
+      expect(() =>
+        render(
+          <Slot>
+            <Text>Hello</Text>
+            <Text>World</Text>
+          </Slot>
+        )
+      ).toThrow('React.Children.only expected to receive a single React element child.');
+    });
+
+    it('renders the child if it is a valid element', () => {
+      render(
+        <Slot>
+          <Text>Hello</Text>
+        </Slot>
+      );
+      expect(screen.getByText('Hello')).toBeInTheDocument();
+    });
+  });
+
+  it('merges styles', () => {
+    render(
+      <Slot style={{color: 'red'}}>
+        <Text style={{backgroundColor: 'blue'}}>Hello</Text>
+      </Slot>
+    );
+    expect(screen.getByText('Hello')).toHaveStyle({
+      color: 'red',
+      backgroundColor: 'blue',
+    });
+  });
+
+  it('merges classNames', () => {
+    render(
+      <Slot className="slot">
+        <Text className="child">Hello</Text>
+      </Slot>
+    );
+    expect(screen.getByText('Hello')).toHaveClass('slot', 'child');
+  });
+
+  it('merges handlers', async () => {
```

Comment:
I would also like to test how `e.preventDefault()` is handled here

Reply:
Would you expect that to be handled? With the Slot, we basically flatten it onto a single DOM node, so imo the handler should fire for both, regardless if preventDefault or stopPropagation is called

Reply:
That's a good point! I guess a manual check for `e.defaultPrevented` would be needed, which seems the least surprising.